src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java

```diff
@@ -1,11 +1,215 @@
 package io.cloudsoft.terraform;
 
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.brooklyn.core.annotation.Effector;
+import org.apache.brooklyn.core.effector.ssh.SshEffectorTasks;
+import org.apache.brooklyn.core.location.Locations;
 import org.apache.brooklyn.entity.software.base.SoftwareProcessImpl;
+import org.apache.brooklyn.feed.function.FunctionFeed;
+import org.apache.brooklyn.feed.function.FunctionPollConfig;
+import org.apache.brooklyn.feed.ssh.SshFeed;
+import org.apache.brooklyn.feed.ssh.SshPollConfig;
+import org.apache.brooklyn.feed.ssh.SshPollValue;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.apache.brooklyn.util.core.task.DynamicTasks;
+import org.apache.brooklyn.util.core.task.system.ProcessTaskStub.ScriptReturnType;
+import org.apache.brooklyn.util.core.task.system.ProcessTaskWrapper;
+import org.apache.brooklyn.util.guava.Maybe;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.brooklyn.util.time.Duration;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 public class TerraformConfigurationImpl extends SoftwareProcessImpl implements TerraformConfiguration {
 
+    private static final Duration FEED_UPDATE_PERIOD = Duration.seconds(30);
+
+    private SshFeed sshFeed;
+
+    private FunctionFeed functionFeed;
+
+    private final AtomicBoolean configurationChangeInProgress = new AtomicBoolean(false);
+
+    private final AtomicBoolean configurationIsApplied = new AtomicBoolean(false);
```

Comment:
If you need these to survive a brooklyn restart then store them in attributes. Will be null on rebind.